### PR TITLE
Functions unified emulator settings

### PR DESCRIPTION
**Pull request by samtstern**
Follow up to #1672 but for `FirebaseFunctions`

### CODE REVIEW

File: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java [hunk 1]
```diff
@@ -47,6 +50,13 @@
 /** FirebaseFunctions lets you call Cloud Functions for Firebase. */
 public class FirebaseFunctions {
 
+  /**
+   * Emulator identifier, see {@link
+   * com.google.firebase.emulators.EmulatorSettings.Builder#addEmulatedService(FirebaseEmulator,
+   * EmulatedServiceSettings)}
+   */
+  public static final FirebaseEmulator EMULATOR = FirebaseEmulator.forName("functions");
```
Comment (samtstern) [inline firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:58]: Should this require a new `api.txt`?  If so the bot did not warn me for the Database PR...

File: firebase-functions/src/androidTest/java/com/google/firebase/functions/CallTest.java [hunk 2]
```diff
@@ -90,7 +90,8 @@ public void testToken() throws InterruptedException, ExecutionException {
             () -> {
               HttpsCallableContext context = new HttpsCallableContext("token", null);
               return Tasks.forResult(context);
-            });
+            },
+            null);
```
Comment (davidmotson) [inline firebase-functions/src/androidTest/java/com/google/firebase/functions/CallTest.java:94]: would it be possible to get a named variable for these nulls? Its unclear what exactly they mean for the test?

Something like "NO_CALLBACK" or something.

File: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java [hunk 3]
```diff
@@ -75,13 +85,26 @@
   private String urlFormat = "https://%1$s-%2$s.cloudfunctions.net/%3$s";
 
   FirebaseFunctions(
-      Context context, String projectId, String region, ContextProvider contextProvider) {
+      Context context,
+      String projectId,
+      String region,
+      ContextProvider contextProvider,
+      @Nullable EmulatedServiceSettings emulatorSettings) {
```
Comment (davidmotson) [inline firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:92]: Is there any value to holding this reference? Or is it only useful for this URL?

File: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java (hunk 1 above)
Comment (davidmotson) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:58]: Unsure, Would it not fail tests/RC confirmation if that was the case?

File: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java (hunk 3 above)
Comment (samtstern) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:92]: Yeah we don't need it after we use it.

File: firebase-functions/src/androidTest/java/com/google/firebase/functions/CallTest.java (hunk 2 above)
Comment (samtstern) [reply firebase-functions/src/androidTest/java/com/google/firebase/functions/CallTest.java:94]: Good idea.

File: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java (hunk 1 above)
Comment (rosariopfernandes) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:61]: It should. Seems like the bot failed silently. See #1692

Comment (samtstern) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:61]: @rosariopfernandes I'll add a fix for that here:
https://github.com/firebase/firebase-android-sdk/pull/1690

Comment (samtstern) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:61]: I think I can add a fix for that here:
https://github.com/firebase/firebase-android-sdk/pull/1690

I don't think there will be a database release before that is resolved.

Comment (rosariopfernandes) [reply firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java:61]: @samtstern I'll change the base branch of my PR to the base branch of #1690

**Pull request merged**